superduperdb/components/schema.py

```diff
@@ -7,25 +7,25 @@
 from superduperdb.components.encoder import Encoder
 
 
-@dc.dataclass
+@dc.dataclass(kw_only=True)
 class Schema(Component):
-    identifier: str
+    type_id: t.ClassVar[str] = 'schema'
+
     fields: t.Mapping[str, t.Union[Encoder, str]]
-    version: t.Optional[int] = None
 
-    type_id: t.ClassVar[str] = 'schema'
+    def __post_init__(self):
+        super().__post_init__()
+
+        assert self.identifier is not None, 'Schema must have an identifier'
```

Comment:
if Component has 
identifier as non optional required

isn't this assertion unnecessary